### PR TITLE
Refactor Geolocator Android

### PR DESCRIPTION
**Pull request by vbuberen**
Updates for Android part to modernise and improve compatibility.

- Removed code related to [pre 1.12 Flutter Android embedding](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects) as it is not supported by recent Flutter versions and min Flutter of geolocator plugin is higher anyway, so no need to save compatibility.
- Removed unused file
- Removed Jetifier as it is not needed in projects that use AndroidX dependencies and because it affects build speed: https://adambennett.dev/2020/08/disabling-jetifier/
- Switched to `ContextCompat` function for broadcast receiver declaration as there were missing flags:
Screenshot 2023-10-09 at 18 44 36

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `main`.
- [x] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

[Contributor Guide]: https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: geolocator_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 4.3.2
+
+- Removes deprecated code from v1 embedding
+- Registers
```
Comment (mvanbeusekom) [inline geolocator_android/CHANGELOG.md:4]: Changelog entries should confirm to the changelog style guide, would you mind updating this to something like this:

```suggestion
- Removes deprecated support for Android V1 embedding from the JAVA code base. Note that the geolocator's Flutter restrictions already require the use of a Flutter version that doesn't support V1 embedding anymore.
```

File: geolocator_android/example/android/gradle.properties [hunk 2]
```diff
@@ -1,3 +1,2 @@
 org.gradle.jvmargs=-Xmx1536M
-android.useAndroidX=true
-android.enableJetifier=true
+android.useAndroidX=true
```
Comment (mvanbeusekom) [inline geolocator_android/example/android/gradle.properties:2]: Nit: missing an empty line at the end of the document.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Hi, 

Thank you very much for this contribution. We are very happy with the changes but have two small styling issues that we would appreciate at fix for.

File: geolocator_android/CHANGELOG.md (hunk 1 above)
Comment (vbuberen) [reply geolocator_android/CHANGELOG.md:4]: Yes, sure. In fact it seems I didn't commit latest changes for changelogs 🤦🏻‍♂️ , so will do it including your feedback.

Comment (vbuberen) [reply geolocator_android/CHANGELOG.md]: Updated

File: geolocator_android/example/android/gradle.properties (hunk 2 above)
Comment (vbuberen) [reply geolocator_android/example/android/gradle.properties]: Updated

**Comment by vbuberen** [comment]
@mvanbeusekom Would like to remind about this PR as I have addressed your feedback, but see that you already released 4.3.2 of the plugin.

**Review by mvanbeusekom — APPROVED** [review]
LGTM, thank you for this contribution

**Pull request merged**